src/components/UserInfo/UserInfoTab/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from 'react';
+import { fetchPosts } from '@/api/user';
+import * as S from './styles';
+import { PostAPIUserInfo } from '@/interfaces/PostAPI';
+import { UserQuizType } from '@/interfaces/UserAPI';
+import UserQuizItem from '../UserQuizItem';
+import TabItem from '../UserInfoTabItem';
+
+function UserInfoTab({ id }: { id: string }) {
+  const [madeQuizzes, setMadeQuizzes] = useState([]);
+  const [commentedQuizzes, setCommentedQuizzes] = useState([]);
+  const [likedQuizzes, setLikedQuizzes] = useState([]);
+
+  const [currentTab, setCurrentTab] = useState(0);
+
+  const tabMapper = [madeQuizzes, commentedQuizzes, likedQuizzes];
+
+  const tabs = [
+    {
+      id: 0,
+      title: '만든 문제',
+    },
+    {
+      id: 1,
+      title: '댓글 단 문제',
+    },
+    {
+      id: 2,
+      title: '좋아요 한 문제',
+    },
+  ];
+
+  useEffect(() => {
+    const updateAllQuiz = async () => {
+      const apiData = await fetchPosts();
+      const realData = apiData.map((post: PostAPIUserInfo) => ({
+        id: post._id,
+        likes: post.likes,
+        comments: post.comments,
+        author: post.author,
+        ...JSON.parse(post.title),
+      }));
+
+      const userMadeQuizzes = realData.filter(
+        (quiz: UserQuizType) => quiz.author._id === id,
+      );
+
+      const userCommentQuizzes = realData.filter((quiz: UserQuizType) => {
+        const userCommentIndex = quiz.comments.findIndex((comment) => {
+          return comment.author._id === id;
+        });
+        const isUserComment = userCommentIndex !== -1;
+        return isUserComment;
+      });
```

Comment:
각 quiz의 코멘트 중 하나라도 id가 일치하는 코멘트가 있다면 해당 퀴즈를 filter하여 반환한다.
some(~)을 통한 filter로직이 설명력을 더 가질 수 있어 제시해보았습니다.
```suggestion
    const userCommentQuizzes = realData.filter((quiz: UserQuizType) => {
        return quiz.comments.some((comment) => comment.author._id === id)
        });

```

Reply:
와.. some은 생각도 못해봤는데 훨씬 이해가 잘되는 코드네요..!! 너무 좋아요.. 제시해주셔서 감사합니다! 바로 반영해 둘게요!! 👍